shm/Ship.cpp

```diff
@@ -0,0 +1,24 @@
+#include "Ship.hpp"
+#include <string>
+
+Ship::Ship()
+    : id_(-1) {}
+
+Ship::Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id)
+    : capacity_(capacity), maxCrew_(maxCrew), crew_(0), speed_(speed), name_(name), id_(id) {}
+    
+Ship::Ship(int maxCrew, int speed, size_t id)
+    : Ship(0, maxCrew, speed, "", id) {}
+
+void Ship::setName(const std::string& name) {
+    name_ = name;
+}
+
+Ship& Ship::operator-=(size_t num) {
+    crew_ -= num;
+    return *this;
+}
+Ship& Ship::operator+=(size_t num) {
+    crew_ += num;
+    return *this;
+}
```

Comment:
What happened when we reach the limit of capacity or drop below 0? You need to handle error cases. Maybe throw exception, maybe return `std::nullopt` or `nullptr` maybe set some specific value like `std::numeric_limits::min/max`.